submissions/Valkirin/frontend-nanodegree-arcade-game/js/app.js

```diff
@@ -0,0 +1,80 @@
+var Enemy = function(x, y, speed) {
+    this.x = x;
+	this.y = y;
+	this.speed = speed;
+    this.sprite = 'images/enemy-bug.png';
+};
+
+Enemy.prototype.update = function(dt) {
+    this.x +=  + (this.speed * dt);
+    
+    if (this.x > 550) {
+    	this.x = -100;
+		this.speed = Math.floor(100 + (Math.random() * 200));
+    };
+    
+    if (player.x > this.x - 60 && 
+        player.x < this.x + 60 && 
+        player.y > this.y - 60 &&
+        player.y < this.y + 60) {
+        player.x = 200;
+        player.y = 400;
+	};
```

Comment:
Referring to global variables breaks OOP principle of encapsulation. Whatever you need from the global scope pass to the class instance via constructor.

Fix here and check all classes not to use globals.

P.S. Global constants are OK to use from inside classes at this stage.

Reply:
fixed